### ISSUE TITLE
Rejected untracked stratum share from daggerhashimoto.usa.nicehash.com

### ISSUE THREAD

**Issue by davembg** [genesismining/sgminer-gm#17]
That's all I'm getting when mining NiceHash
I have the no-extranonce set to false, and the stratum URL ends with /#xnsub

Am I missing something?

The url is:
"url": "stratum+tcp://daggerhashimoto.usa.nicehash.com:3353/#xnsub",

**Comment by tpruvot** [comment]
extranonce = nicehash, so i guess it should not be set to false ;)

**Comment by davembg** [comment]
I tried setting "--no-extranonce" to false/true etc,
--no-extranonce = true DISABLES it, right?
Anyway, I tried both true/false options, and removed it all together - no joy.

**Comment by tpruvot** [comment]
yep, its the fight beween nicehash and GM about this option ;) lol boolean options are a bit weird, its --no-extranonce without flag on the command line or no flag at all

**Comment by davembg** [comment]
Yeah, I attempted to support their proto on donkeypool.com. I didn't try very hard😀

**Comment by tpruvot** [comment]
try to comment that : https://github.com/genesismining/sgminer-gm/blob/master/sgminer.c#L623-L628

**Comment by davembg** [comment]
Thx, I'll give it a go and post results.

**Comment by davembg** [comment]
Yeah, !disabling extranonce is probably closer.
The stratum messages aren't quite NiceHash friendly - I'll look at it later.

I don't want to do a PR because it looks to me like Wolf0 and GM don't really want this miner to support NiceHash. I'd like to respect that....

**Comment by wolf9466** [comment]
@davembg No, not true at all. You may notice we have a pull request open to incorporate NiceHash's features. It'll come in soon. @wolf9466, @NaN-git and I will be out of the office for a little bit for Christmas, but we'll work on this once we have a moment during the holidays.

Merry Christmas, @davembg!

And always submit pull requests. Don't ever fret about that. We won't bite. =)

**Comment by OhGodAGirl** [comment]
@davembg No, not true at all. You may notice we have a pull request open to incorporate NiceHash's features. It'll come in soon. @wolf9466, @NaN-git and I will be out of the office for a little bit for Christmas, but we'll work on this once we have a moment during the holidays.

Merry Christmas, by the way, @davembg!

And always submit pull requests. Don't ever fret about that. We won't bite. Worst case, we'd politely decline code and provide a _very_ good reason for doing so.

**Comment by davembg** [comment]
HA! I noticed the PR - AFTER I WAS ALMOST DONE "Fixing" it.
@tpruvot pointed me in the right direction. If I have time, I'll clone the repo and do a PR.
Might save some time - the changes are trivial.

@wolf9466 Merry Christmas to you too bud!

**Comment by VRImage** [comment]
Hey. ever get it to work?
Have the same issue and can't seem to get it fixed.

**Comment by davembg** [comment]
Not really. I got caught-up in the holiday and never moved back to it.
Sorry about that.
